test/fs/test_fs_readdir_ino_matches_stat_ino.c

```diff
@@ -0,0 +1,50 @@
+#include <stdio.h>
+#include <unistd.h>
+#include <assert.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <assert.h>
+#include <errno.h>
+#include <string.h>
+#include <dirent.h>
+
+
+int main() {
+  int res = open("b", O_CREAT, 0777);
+  assert(res >= 0);
+  assert(close(res) == 0);
+  assert(symlink("b", "a") == 0);
```

Comment:
Maybe put there 4 lines in a `setup` function since they effectively are preparing the directory.